react-js/react-part4-full-apps.md

```diff
@@ -0,0 +1,343 @@
+
+```sh
+create-react-app my-fun-app
+```
+
+Since the main benefit of CRA is that you don't need to worry about how it works, we're mainly going to focus on trying it out through guided practice but there are links below if you want to know more about how it works.
+
+### Common Mistakes & Misconceptions
+
+- CRA is only for toy apps and _real engineers_ setup React from scratch
+  - Actually, many professional projects leverage CRA or another starting point template like [Next.js](https://nextjs.org/) when beginning applications. Even if you want to customize the app in some ways, the general setup is usually a good starting point for many types of projects and **it will not make your portfolio project seem amateur to use it**
+- You **must** use CRA to build React apps
+  - Not true, but it is strongly recommended you build experience using it first before trying to do it yourself without CRA. Although it's much easier, especially for a beginner, many companies have ofted to create a customized setup so they have full control over all aspects of the application
+
+## Guided Practice
+
+Let's setup a small app.
+
+- `npx create-react-app todo-cra`
+- _grab a coffee while it installs everything_
+- `cd todo-cra`
+- `yarn start`
+
+The app should open in a new browser tab automatically and show a spinning atom logo.
+
+- `code .` to open the generated project in your editor
+- Find `App.js` in the `src` folder
+- Change the contents of `<p>Hello Techtonica!</p>` and save
+- The browser should live update automatically 🥳
+
+### Folder Structure
+
+Review the [existing files](https://create-react-app.dev/docs/folder-structure) of the project.
+
+The most important files are
+
+1. `src/App.css` - all the styles for your React components
+1. `public/index.html` - the non-React content of your web pages, e.g. favicon, static footer, etc
+1. `src/App.js` - mounts your root component
+
+#### `App.css`
+
+Change the color of the "Learn React" link to `#b36ff6` or another color you like on https://colors.lol/. Changes to CSS will live update as well.
+
+#### `index.html`
+
+- Change the page title to "TODO"
+- Find a nice favicon on https://www.flaticon.com/
+- Make this the new favicon
+
+Now let's add the main application logic to `App.js`. We'll keep everything in one file for now but as your app grows, you probably want to split components to their own files.
+
+### Starter Components
+
+Create a new file.
+
+```jsx
+// Todo.js
+import React from 'react';
+
+class Todo extends React.Component {
+  render() {
+    const { todo } = this.props;
+    return <div>{todo.text}</div>;
+  }
+}
+export default Todo;
+```
+
+```jsx
+// App.js
+import React from 'react';
+import Todo from './Todo';
+import './App.css';
+
+class Todo extends React.Component {
+  render() {
+    const { todo } = this.props;
+    return <div>{todo.text}</div>;
+  }
+}
+
+class App extends React.Component {
+  state = {
```

Comment:
I'm afraid state isn't this simple...

Reply:
https://reactjs.org/docs/react-component.html#constructor

Reply:
Oops good catch! I ran the code but might've "improved" it after pasting it here.